src/Appwrite/Platform/Workers/Webhooks.php

```diff
@@ -117,10 +133,87 @@ private function execute(array $events, string $payload, Document $webhook, Docu
+            } else {
+                $logs .= 'Status code: ' . $statusCode . "\n";
+                $logs .= 'Body: ' . "\n" . \mb_strcut($responseBody, 0, 10000) . "\n"; // Limit to 10kb
+            }
+
+            $webhook->setAttribute('logs', $logs);
+
+            if ($attempts >= self::MAX_FAILED_ATTEMPTS) {
+                $webhook->setAttribute('enabled', false);
+
+                $memberships = $dbForConsole->find('memberships', [
+                    Query::equal('teamInternalId', [$project->getAttribute('teamInternalId')]),
+                    Query::limit(APP_LIMIT_SUBQUERY)
+                ]);
+
+                $userIds = array_column(\array_map(fn ($membership) => $membership->getArrayCopy(), $memberships), 'userId');
+
+                $users = $dbForConsole->find('users', [
+                    Query::equal('$id', $userIds),
+                ]);
+
+                $protocol = App::getEnv('_APP_OPTIONS_FORCE_HTTPS') == 'disabled' ? 'http' : 'https';
+                $hostname = App::getEnv('_APP_DOMAIN');
+                $projectId = $project->getId();
+                $webhookId = $webhook->getId();
+
+                $template = Template::fromFile(__DIR__ . '/../../../../app/config/locale/templates/email-webhook-failed.tpl');
+
+                $template->setParam('{{webhook}}', $webhook->getAttribute('name'));
+                $template->setParam('{{project}}', $project->getAttribute('name'));
+                $template->setParam('{{url}}', $webhook->getAttribute('url'));
+                $template->setParam('{{error}}', $curlError ??  'The server returned ' . $statusCode . ' status code');
+                $template->setParam('{{redirect}}', $protocol . '://' . $hostname . "/console/project-$projectId/settings/webhooks/$webhookId");
+                $template->setParam('{{attempts}}', $attempts);
+
+                $subject = 'Webhook deliveries have been paused';
+                $body = Template::fromFile(__DIR__ . '/../../../../app/config/locale/templates/email-base-styled.tpl');
+
+                $body
+                    ->setParam('{{subject}}', $subject)
+                    ->setParam('{{message}}', $template->render())
+                    ->setParam('{{year}}', date("Y"));
+
+                $queueForMails
+                    ->setSubject($subject)
+                    ->setBody($body->render());
+
+                foreach ($users as $user) {
+                    $queueForMails
+                        ->setVariables(['user' => $user->getAttribute('name', '')])
+                        ->setName($user->getAttribute('name', ''))
+                        ->setRecipient($user->getAttribute('email'))
+                        ->trigger();
+                }
+            }
+
+            $dbForConsole->updateDocument('webhooks', $webhook->getId(), $webhook);
+            $dbForConsole->deleteCachedDocument('projects', $project->getId());
+
+            $this->errors[] = $logs;
```

Comment:
what is this attribute used for?

Reply:
No idea. It was part of the already existing code, so I didn't touch this bit. 😅